app/controllers/catalog_controller.rb

```diff
@@ -41,6 +42,48 @@ def update_user_per_cat_page
     end
   end
 
+  def submit_form # rubocop:disable MethodLength, AbcSize
```

Comment:
I'm ok with tweaking the cart date handling through this issue so that we can use the same code for both use cases. Don't forget, this issue is designed to allow for cart *edits* through the checkout page, so there really isn't such a thing as the "original" cart - we're meant to be editing the *actual* cart by submitting the form. If a user enters invalid dates here I think we'd want it to behave the way it would in the catalog views; that keeps us consistent.

That's just my opinion, though; let me know if that makes sense or if you still disagree :smile:.